### PR TITLE
[EPROD-668] Get chunks number from UnboundedMap

### PR DESCRIPTION
**Pull request by F3kilo**
https://infinityswap.atlassian.net/browse/EPROD-668

### CODE REVIEW

File: ic-stable-structures/src/structure/stable_storage/unbounded.rs [hunk 1]
```diff
@@ -62,6 +62,20 @@ where
         StableUnboundedIter(self.inner.iter().peekable())
     }
 
+    /// Returns number of chunks, used to store the entity.
+    pub fn chunks_number_of(&self, key: &K) -> Option<ChunkIndex> {
+        let first_chunk_key = Key::new(key);
+        let max_chunk_key = first_chunk_key.clone().with_max_chunk_index();
+
+        // first item in this iter will be the last chunk of the key.
+        self.inner
```
Comment (ufoscout) [inline ic-stable-structures/src/structure/stable_storage/unbounded.rs:71]: How much expensive is this? It seems to me that this requires to decode the entire data

**Review by ufoscout — CHANGES_REQUESTED** [review]

File: ic-stable-structures/src/structure/stable_storage/unbounded.rs (hunk 1 above)
Comment (F3kilo) [reply ic-stable-structures/src/structure/stable_storage/unbounded.rs:71]: It encodes the `key`, and performs one binary search in the map. The search requires reading (without decoding)  log_2(N) keys.

**Review by veeso — APPROVED** [review]

**Review by ufoscout — COMMENTED** [review]
Waiting for the changes we discussed offline to be implemented

**Review by ufoscout — CHANGES_REQUESTED** [review]
Waiting for the changes we discussed offline to be implemented

**Review by GraDKh — APPROVED** [review]

**Comment by F3kilo** [comment]
> Waiting for the changes we discussed offline to be implemented

Already done. There is the `total_chunks_number()` method. 
I'll keep the `chunks_number_of()` method too. It could be useful in another task.

**Comment by ufoscout** [comment]
> I'll keep the `chunks_number_of()` method too. It could be useful in another task.

@F3kilo I would not keep it instead, it has hidden performance implications which are the reason why we implemented the alternative solution.

**Review by ufoscout — CHANGES_REQUESTED** [review]
See my comments

**Comment by F3kilo** [comment]
> @F3kilo I would not keep it instead, it has hidden performance implications which are the reason why we implemented the alternative solution.

Done

**Pull request merged**